include/common/types.h

```diff
@@ -157,6 +157,24 @@ class ValTypeBase {
   } Inner;
 };
 
+/// HeapType definition. The RefType is the subset of the RefType.
+class HeapType : public ValTypeBase {
+public:
+  HeapType() noexcept = default;
+  // Constructor for the heap types (func and extern).
+  HeapType(HeapTypeCode HT) noexcept
+      : ValTypeBase(ValTypeCode::RefNull, HT, 0) {
+    assuming((Inner.Data.Code == ValTypeCode::RefNull));
+    assuming((Inner.Data.HTCode != HeapTypeCode::TypeIndex) &&
+             (Inner.Data.HTCode != HeapTypeCode::NotHeapType));
+  }
+  // Constructor for the heap types (type index).
+  HeapType(uint32_t I) noexcept
+      : ValTypeBase(ValTypeCode::RefNull, HeapTypeCode::TypeIndex, I) {
```

Comment:
I prefer to reserve the originial design here (hidden the `HeapType` and use `RefType` directly).
Because the `HeapType` doesn't describe the `RefType`, the reference type may be `Ref` or `RefNull`, not always `RefNull`.

Reply:
Alright. I didn't realize the GC proposal. Thanks.